src/lib/utils/router.js

```diff
@@ -193,6 +195,9 @@ export function route(url) {
   globalHandler(candidateUrl, u.hash).then((aborted) => {
     if (!aborted) {
       scrollToHashOrTop(u.hash);
+      addPageToContentIndex(u.href).catch((error) => {
```

Comment:
@samthor, following up from https://github.com/GoogleChrome/web.dev/pull/2798/files#r420448496, this is my attempt to add to the content index on the fly.

I am not confident that this is the right place to put this code, but _somewhere_ within `router.js` seemed like a reasonable place to put some code that is meant to be run following user navigation. If there's a better location for it, though, I'm happy to put it elsewhere.

I'm also not 100% sure that at this execution point the new page's JSON is guaranteed to be in the Cache Storage API, so if you have doubts about that, we should move it to someplace where it would run later in the navigation lifecycle, when everything is fully cached.